### PR TITLE
CSharp and Java versions for the-state-machine stack. Update TS and Python Stack to CDK 1.82.0. Documentation issues.

### PR DESCRIPTION
**Pull request by leandrodamascena**
Good evening. In this PR I created the C# and Java version and made some changes in TS/Python stacks. These changes include a refactored piece of code due to being obsolete/deprecated.

1 - TS and Python versions updated to CDK 1.82.0.
2 - Fixed some comments. Just to keep the things organized.

Let me know if I need to change anything.

Ty

### CODE REVIEW

File: the-state-machine/typescript/lib/the-state-machine-stack.ts [hunk 1]
```diff
@@ -16,16 +16,19 @@ export class TheStateMachineStack extends cdk.Stack {
...
+      code: lambda.Code.fromAsset('lambda-fns'),     // code loaded from the "lambda-fns" directory
       handler: 'orderPizza.handler'           // file is "orderPizza", function is "handler"
     });
 
     // Step functions are built up of steps, we need to define our first step
-    const orderPizza = new sfn.Task(this, 'Order Pizza Job', {
-      task: new tasks.InvokeFunction(pineappleCheckLambda),
+    // Refactored code because sfn.Task is Depecrated right now. Using StepFunctionsTaks.LambdaInvoke instead
+    // https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html
+    const orderPizza = new tasks.LambdaInvoke(this, "Order Pizza Job", {
```
Comment (leandrodamascena) [inline the-state-machine/typescript/lib/the-state-machine-stack.ts:26]: This piece of code was obsolete (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html). I changed it to the new implementation using service as invocation (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions-tasks.LambdaInvoke.html)

File: the-state-machine/python/the_state_machine/the_state_machine_stack.py [hunk 2]
```diff
@@ -17,17 +17,19 @@ def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
...
 
         # Step functions are built up of steps, we need to define our first step
-        order_pizza = step_fn.Task(self, 'Order Pizza Job',
-                                   task=step_fn_tasks.InvokeFunction(pineapple_check_lambda),
-                                   input_path='$.flavour',
-                                   result_path='$.pineappleAnalysis'
-                                   )
+        # Refactored code because sfn.Task is Depecrated right now. Using StepFunctionsTaks.LambdaInvoke instead
+        # https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html
+        order_pizza = step_fn_tasks.LambdaInvoke(self, 'Order Pizza Job',
```
Comment (leandrodamascena) [inline the-state-machine/python/the_state_machine/the_state_machine_stack.py:28]: This piece of code was obsolete (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html). I changed it to the new implementation using service as invocation (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions-tasks.LambdaInvoke.html)

**Comment by nideveloper** [comment]
The step function definitions fluctuated A LOT last year, I think it is stable now though. I do know that this change breaks the typescript unit tests since the step function definition is different but testing it post deploy the behaviour is the same so the unit test just needs updated for the new definition from this new lambdaInvoke method

**Comment by leandrodamascena** [comment]
Hmm you're right! Tbh, when I refactor or add new languages ​​to the stacks, I redeploy all stacks in all languages and check if it is working as expected, but I don't run the ts tests, that's a failure, my bad! I will fix and update the pr.

**Comment by nideveloper** [comment]
This made me wonder why my github action didn't fail which runs all the ts unit tests and cdk synths all the python versions. There was a bug in the ts script in that it was running all the tests but the exit on error flag wasn't set so when tests failed it reported that the build still passed. I just fixed that and some broken tests with my latest PR so some of the patterns you have updated might have had broken ts tests that I have now fixed. The test in this pr aside, it's worth checking if you run the ts tests in any of your branches and they fail if I have already fixed it as all tests now pass

**Comment by leandrodamascena** [comment]
Hmm tomorrow I will check this PR and yours considerations!  At a glance I saw that the unit tests did not break, even changing constructor, but I need to analyze in detail.

**Comment by nideveloper** [comment]
All you need to do is on your branch cd into the-state-machine/typescript, run npm i then npm run build && npm run test you should see a test failure:

testfail

**Comment by leandrodamascena** [comment]
Yep, my tests were cached and jest couldn't test again, i have no idea why but I cleaned up here and now it's working.

Once the lambda invokation has changed, now the synth includes a new "ErrorEquals" parameters with lambda exceptions and, because of this, the previous test failed.

Now I think it is ok.

**Comment by leandrodamascena** [comment]
image

**Pull request merged**